Whiskey/Tasks/Invoke-WhiskeyNpmConfig.ps1

```diff
@@ -0,0 +1,133 @@
+
+function Invoke-WhiskeyNpmConfig
+{
+    <#
+    .SYNOPSIS
+    Runs the `npm config` command to set global-level, user-level, or project-level NPM configuration.
+
+    .DESCRIPTION
+    The `NpmConfig` task runs the `npm config` commadn to set NPM configuration. By default, it sets configuration at the project level (i.e. in a .npmrc file in the current directory). To set the configuration at the user-level, set the `Scope` property to `User`. To set the configuration globally, set the `Scope` property to `Global`.
+
+    Set the `Configuration` property to key/value pairs. The keys should be the name of the configuration setting; the value should be its value. For example,
+
+        BuildTasks:
+        - NpmConfig:
+            Configuration:
+                registry: https://registry.npmjs.org
+                email: buildmaster@example.com
+     
+    # Properties
+
+    * `Configuration`: the key/value pairs to set.
+    * `Scope`: the level at which the configuration is set. By default (i.e. if this property is missing), the configuration is set at the project leel, in an .npmrc file in the current directory. Allowed value are `Projec`, `User`, or `Global` to set the configuration in the project, user, or global  NPM configuration files, respectively.
+    * `WorkingDirectory`: the directory in which to run the `npm config` command. The default is the directory of your whiskey.yml file.
+
+    # Examples
+
+    ## Example 1
+
+        BuildTasks:
+        - NpmConfig:
+            Configuration:
+                registry: https://registry.npmjs.org
+                email: buildmaster@example.com
+
+    Will create a .npmrc file in the current directory that looks like this:
+
+        registry=https://registry.npmjs.org
+        email=buildmaster@example.com
+        
+    ## Example 2
+
+        BuildTasks:
+        - NpmConfig:
+            Configuration:
+                registry: https://registry.npmjs.org
+                email: buildmaster@example.com
+            Scope: User
+
+    Will create a .npmrc file in the user's home directory that looks like this (assuming the user's .npmrc file doesn't exist):
+
+        registry=https://registry.npmjs.org
+        email=buildmaster@example.com
+    #>
+    [Whiskey.Task('NpmConfig')]
+    [Whiskey.RequiresTool('Node','NodePath')]
+    param(
+        [Parameter(Mandatory=$true)]
+        [object]
+        $TaskContext,
+
+        [Parameter(Mandatory=$true)]
+        [hashtable]
+        $TaskParameter
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    $nodePath = $TaskParameter['NodePath']
+    if( -not $nodePath -or -not (Test-Path -Path $nodePath -PathType Leaf) )

```

Comment:
It's me being a little paranoid. I'd rather give the user an explicit error message about what might be wrong then a "path is null or empty" message.

Reply:
In that case, there are a few tasks, such as `NpmInstall` and maybe others, that are missing this check to ensure `$nodePath` exists. In case you wanted to go back and add them.